MapboxCoreNavigation/RouteController.swift

```diff
@@ -719,23 +721,31 @@ extension RouteController: CLLocationManagerDelegate {
                 
                 // Look at the following step to determine what the new alert level should be
                 if let upComingStep = routeProgress.currentLegProgress.upComingStep {
-                    alertLevel = upComingStep.expectedTravelTime <= RouteControllerMediumAlertInterval ? .medium : .low
+                    alertLevel = calculateNewAlert(from: upComingStep)
                 } else {
                     assert(false, "In this case, there should always be an upcoming step")
                 }
             }
-        } else if secondsToEndOfStep <= RouteControllerHighAlertInterval && routeProgress.currentLegProgress.currentStep.distance > minimumDistanceForHighAlert {
+        } else if secondsToEndOfStep <= RouteControllerHighAlertInterval {
             alertLevel = .high
-        } else if secondsToEndOfStep <= RouteControllerMediumAlertInterval &&
-            // Don't alert if the route segment is shorter than X
-            // However, if it's the beginning of the route
-            // There needs to be an alert
-            routeProgress.currentLegProgress.currentStep.distance > minimumDistanceForMediumAlert {
+        } else if secondsToEndOfStep <= RouteControllerMediumAlertInterval {
             alertLevel = .medium
         }
         
         incrementRouteProgress(alertLevel, location: location, updateStepIndex: updateStepIndex)
     }
+    
+    func calculateNewAlert(from upcomginStep: RouteStep) -> AlertLevel {
+        let expectedTravelTime = upcomginStep.expectedTravelTime
+        switch expectedTravelTime {
+        case let x where x <= RouteControllerHighAlertInterval:
```

Comment:
This is a pretty convoluted way of writing an if/else if/else statement. To fully take advantage of switch/case, use ranges like this:

```swift
switch expectedTravelTime {
case 0..<RouteControllerHighAlertInterval: …
case RouteControllerHighAlertInterval..<RouteControllerMediumAlertInterval: …
default: …
}
```